src/Message/Handler/AbstractEntityHandler.php

```diff
@@ -22,7 +22,7 @@ public function __construct(ManagerRegistry $managerRegistry, private readonly I
         $this->managerRegistry = $managerRegistry;
     }
 
-    public function __invoke(RemoveEntity|IndexEntity $message): void
+    protected function handle(RemoveEntity|IndexEntity $message): void
```

Comment:
```suggestion
    protected function handle(RemoveEntity|IndexEntity $message, callable $action): void
```

I think I would change the signature to this and add the callback in the concrete handler like:

```php
$this->handle($message, fn(Index $index) => $index->indexer()->removeEntity($entity););
```

Wdyt?